django_project/trail/models/trail.py

```diff
@@ -0,0 +1,118 @@
+
+# coding=utf-8
+"""Model definitions for a trail app"""
+import os
+
+from django.contrib.gis.db import models
+from django.utils.translation import ugettext_lazy as _
+from django.utils.text import slugify
+from django.core.urlresolvers import reverse
+from unidecode import unidecode
+
+from core.settings.contrib import STOP_WORDS
+from django.conf.global_settings import MEDIA_ROOT
+
+__author__ = 'Alison Mukoma <alison@kartoza.com>'
+__date__ = '01/09/2018'
+__license__ = "GPL"
+__copyright__ = 'kartoza.com'
+
+
+def increment_slug(_name, _colour):
+    """Function to increment slug.
+
+    If there already exists the trail name and colour,
+    the slug will be incremented
+    e.g. trail-#25HHfff, trail-#25HHfff etc.
```

Comment:
I dont thing it makes sens to use colour for the slug - rather trail name...

Reply:
also this is the third time I see the same (or a very similar function). Keep your code DRY. (Do not Repeat Yourself) [I just repeated this  :P ]